src/VisualStudio/Core/Def/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -310,8 +305,8 @@ public async ValueTask ApplyAsync(IVsAsyncFileChangeEx2 service, CancellationTok
                     var cookie = await service.AdviseDirChangeAsync(_paths[0], watchSubdirectories: true, _sink, cancellationToken).ConfigureAwait(false);
                     _cookies.Add(cookie);
 
-                    if (_filter != null)
-                        await service.FilterDirectoryChangesAsync(cookie, [_filter], cancellationToken).ConfigureAwait(false);
+                    if (_filters != null)
```

Comment:
Did you meant to check for non-empty, not non-null/default if this is an ImmutableArray? I imagine this will work no matter what (FilterDirectoryChangesAsync says an empty array means no filter) but it's either an unnecessary call, or an unnecessary check.

Reply:
That was an oops when converting from a nullable type to an ImmutableArray. Good catch!